

# Exploring and Annotating Variants using Bioconductor
Chela James (Adapted from Francesco Lescai 2012)
Institute of Child Health
University College London
c.james@ucl.ac.uk
February 16, 2016

## Introduction

This tutorial is intended to give you an overview of the workflow commonly employed for data generated by parallel DNA sequencing.  Specifically, we will examine data from data from the coding regions of genome, i.e. the exome.  The initial steps of the process are computationally intensive and require some time to be executed so they will be described. But the final steps â€“ annotation and analysis can be done in R.


## Aligning Sequence Data
The file we receive following sequencing is known as a fastq, it typically consists of millions of lines but each sequence read is encoded by four lines
```
@SEQ_ID
GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT
+
!''*((((***+))%%%++)(%%%%).1***-+*''))**55CCF>>>>>>CCCCCCC65
```

The id line is unique for each read and it tells us which machine and lane the read comes from and whether it is the first or second of a pair of reads. The second is the sequence of the read, the third a spacer and the fourth an ascii code for the quality of each base call of the sequence read.


```
  SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS.....................................................
  ..........................XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX......................
  ...............................IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII......................
  .................................JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ......................
  LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL....................................................
  !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
  |                         |    |        |                              |                     |
 33                        59   64       73                            104                   126
  0........................26...31.......40                                
                           -5....0........9.............................40 
                                 0........9.............................40 
                                    3.....9.............................40 
  0.2......................26...31........41                              

 S - Sanger        Phred+33,  raw reads typically (0, 40)
 X - Solexa        Solexa+64, raw reads typically (-5, 40)
 I - Illumina 1.3+ Phred+64,  raw reads typically (0, 40)
 J - Illumina 1.5+ Phred+64,  raw reads typically (3, 40)
     with 0=unused, 1=unused, 2=Read Segment Quality Control Indicator (bold) 
     (Note: See discussion above).
 L - Illumina 1.8+ Phred+33,  raw reads typically (0, 41)
 ```
The fastq is aligned usually using a high performance computer or cluster using software such as novoalign, stampy or BWA, this software takes each of these reads which are randomly distributed in the fastq and aligns to the reference sequence (in the case of the human reference ~3000 000 000 nucleotides.


The output is a sam file or its equivalent compressed bam file
<a href="https://samtools.github.io/hts-specs/SAMv1.pdf">Full specification for SAM here</a>


```
NB501092:15:H5JJLAFXX:1:11101:4282:1085	81	2	16080075	60	80M	=	16079505	-650	TTCAATTTGAAACTGGAAACATCCAGAGGTCTTGTTCCTAAGGGGGCGCGGCCGCTCCCTGCTATTTTGCACCTTCGGAC	/EEEE6EE/EEE/EA/6EEE/EEEEAEEEEEE6E/EEEEEEEEEEEEAEE/EE6EEEEEEEEEEE6EEEEEEEEEAA//A	NM:i:2	MD:Z:50T2T26	AS:i:70	XS:i:19	RG:Z:ping_test
```

The file format is perhaps easier to understand if it is visualised with can be done with the integrated genomics view IGV again developed by the Broad instititue

<a href="https://basespace.files.wordpress.com/2012/10/igv_screenshot1.png">bam file</a>


## Calling variants with GATK

Following alignment, we then call those regions of the test sequence that differ from the reference, there are various variat callers but we will discuss the <a href="https://www.broadinstitute.org/gatk/">Genome Analysis Toolkit</a> (GATK) pipeline.  The GATK pipeline is developed at the Broad Institute in the US.
The challenge of variant calling is to accurately report those real differences from the reference sequence with the aim of filtering those changes that might be significant from common or insignificant ones.

The GATK pipeline involves realigning and recalibrating the bam files generated by BWA and then calling the variants

Variants known as indels are notoriously difficult to identify with parallel sequencing because of the shortness of the reads, realignment with GATK improves this somewhat by realigning around known indels for example this <a href="https://raw.githubusercontent.com/mozack/abra/master/misc/example.png">alignment</a> is massively improved by realignment.  Recalibration can also improve the confidence in variant calls because of it removes <a href="http://cdn.vanillaforums.com/gatk.vanillaforums.com/FileUpload/16/257337235569ff4ea8f4a05e803b8c.png">technical variations in dinucleotides</a>

We routinely call the variants the GATK haplotype caller tools.   This is a sophisticated piece of software which identifies so called active regions and uses a  
